### PR TITLE
Add timer to clean deadlined executions

### PR DESCRIPTION
**Pull request by mcodato**
Hi @Al2Klimov
in this PR we added a timer to clean up the deadlined executions

### CODE REVIEW

File: lib/icinga/checkable.cpp [hunk 1]
```diff
@@ -257,3 +263,34 @@ void Checkable::ValidateMaxCheckAttempts(const Lazy<int>& lvalue, const Validati
...
+		if (executions) {
+			for (const String& key : executions->GetKeys()) {
+				execution = executions->Get(key);
+				if (execution->Contains("deadline") && now > execution->Get("deadline")) {
+					executions->Remove(key);
+				}
+			}
+		}
+	}
+}
```
Comment (Al2Klimov) [inline lib/icinga/checkable.cpp:296]: A newline at EOF is missing.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by mcodato** [comment]
# TEST

### Configuration

**/etc/icinga2/conf.d/test.conf**

```
object Host "satellite" {
    address = "127.0.0.1"
    check_command = "hostalive"
    zone = "satellite"
    vars.ls_dir = "/tmp/host"
    enable_active_checks = false
}

object Host "agent" {
    address = "127.0.0.1"
    check_command = "hostalive"
    zone = "agent"
    vars.ls_dir = "/tmp/host"
    enable_active_checks = false
}

object Host "master" {
    address = "127.0.0.1"
    check_command = "hostalive"
}

object CheckCommand "custom_command" {
    command = [ "/usr/bin/ls", "$ls_dir$" ]
    zone = "global-commands"
}

apply Service "custom_service" {
    check_command = "custom_command"
    enable_active_checks = false
    assign where host.address
    event_command = "event_command_echo"
}

apply Service "ping" {
    check_command = "ping"
    assign where host.address
    zone = "master"
}

```

**/etc/icinga2/zones.conf**
```
object Zone "global-commands" {
    global = true
}

object Zone "global-templates" {
    global = true
}

object Zone "director-global" {
    global = true
}

object Endpoint "master" {
}

object Zone "master" {
    endpoints = [ "master" ]
}

object Endpoint "satellite" {
}

object Zone "satellite" {
    endpoints = [ "satellite" ]
    parent = "master"
}

object Endpoint "agent" {
    log_duration = 0s
}

object Zone "agent" {
    endpoints = [ "agent" ]
    parent = "satellite"
}

```

### Request
```
curl -v -k -s -u root:admin -H 'Accept: application/json' -X POST 'https://localhost:5665/v1/actions/execute-command' -d '{"type":"Service", "service": "master!ping", "ttl":"15", "macros": {"command_endpoint": "master"}, "command": "ping", "command_type": "CheckCommand" }'
```

### Response
```
{"results":[{"checkable":"master!ping","code":202.0,"execution":"3af74bbe-8948-4f52-ae01-22ca2a1b6595","status":"Accepted"}]}
```

### Logs
```
[2020-08-26 17:24:14 +0200] information/ApiListener: New client connection from [127.0.0.1]:51554 (no client certificate)
[2020-08-26 17:24:14 +0200] notice/ApiListener: New HTTP client
[2020-08-26 17:24:14 +0200] information/HttpServerConnection: Request: POST /v1/actions/execute-command (from [127.0.0.1]:51554), user: root, agent: curl/7.66.0).
[2020-08-26 17:24:14 +0200] debug/HttpUtility: Request body: '{"type":"Service", "service": "master!ping", "ttl":"15", "macros": {"command_endpoint": "master"}, "command": "ping", "command_type": "CheckCommand" }'
[2020-08-26 17:24:14 +0200] notice/ApiActionHandler: Running action execute-command
[2020-08-26 17:24:14 +0200] notice/ApiListener: Relaying 'event::UpdateExecutions' message
[2020-08-26 17:24:14 +0200] critical/ApiListener: Not implemented.
[2020-08-26 17:24:14 +0200] information/HttpServerConnection: HTTP client disconnected (from [127.0.0.1]:51554)
```

### Console
```
<1> => var s = get_service("master","ping");
null
<2> => s.executions
{
	"3af74bbe-8948-4f52-ae01-22ca2a1b6595" = {
		deadline = 1598455536.734377
		pending = true
	}
}

```

After 5 minutes
```
<3> => s.executions
{
}
```

**Comment by Al2Klimov** [comment]
@htriem Verify this after the other one.

**Comment by htriem** [comment]
After recreating the test with the supplied config and testing this PR on the whole, everything works as expected. 👍

**Pull request merged**